src/types/index.ts

```diff
@@ -10,6 +10,8 @@ type GroundPlacesFile = Record<Gpuid, GroundPlaceFromFile>;
 
 type GroundPlace = StopGroup | StopCluster;
 
+type GroundPlaceDiff = Record<Gpuid, GroundPlaceDiffOptions>;
```

Comment:
Do this respect the order ?

Reply:
If so AMAZING, if NOT --> bad

Reply:
Je range les types par catégorie, et j'essaye dans chaque catégorie de les classer par ordre si c'est ca dont tu parle ?

Par exemple le dernier type de la liste, ici GroundPlaceActionHistory, utilise un record avec GroundPlaceActionOptions qui est la dernière interface de sa catégorie

Reply:
je pense que Simon parle de l'ordre "chronologique", genre si tu as move puis delete un stopGroup, il faut que ce soit bien "rangé" pour les parser dans le bon ordre (move puis delete)

Reply:
Au niveau des actions, il n'y a pas besoin de créer un ordre je pense

Par exemple l'action `createStopGroup` va elle même appeler les actions nécessaires pour effectuer un bon `create`.

C'est ici qu'intervient la possibilité de ne pas créer une `ActionHistory`. Vu que `createStopGroup` génère son propre **ActionHistory**, au sein de la méthode `createStopGroup`, les autres méthodes qui peuvent être appelées n'en généreront pas

Je ne sais pas si c'est le point que tu voulais soulever ?

Reply:
Oui, pas de soucis de ce  côté (on a fait un point dessus avec @theo-le-donnee c'est bien un tableau de **GroundPlaceActionHistory** (où le record est tout simplement un objet comportant l'action). Et il ne risque pas d'y avoir de soucis de chronologie, car toutes les actions sont poussées à la suite des autres (array.push)